contrib/systemd/etcd2-backup-coreos/etcd2-restore.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"regexp"
+	"time"
+)
+
+var etcdctlPath,
+	etcdPath,
+	etcdRestoreDir,
+	etcdName,
+	etcdPeerUrls string
+
+func main() {
+	flag.StringVar(&etcdctlPath, "etcdctl-path", "/usr/bin/etcdctl", "absolute path to etcdctl executable")
+	flag.StringVar(&etcdPath, "etcd-path", "/usr/bin/etcd2", "absolute path to etcd2 executable")
+	flag.StringVar(&etcdRestoreDir, "etcd-restore-dir", "/var/lib/etcd2-restore", "absolute path to etcd2 restore dir")
+	flag.StringVar(&etcdName, "etcd-name", "default", "name of etcd2 node")
+	flag.StringVar(&etcdPeerUrls, "etcd-peer-urls", "", "advertise peer urls")
+
+	flag.Parse()
+
+	if etcdPeerUrls == "" {
+		panic("must set -etcd-peer-urls")
+	}
+
+	if finfo, err := os.Stat(etcdRestoreDir); err != nil {
+		panic(err)
+	} else {
+		if !finfo.IsDir() {
+			panic(fmt.Errorf("%s is not a directory", etcdRestoreDir))
+		}
+	}
+
+	if !path.IsAbs(etcdctlPath) {
+		panic(fmt.Sprintf("etcdctl-path %s is not absolute", etcdctlPath))
+	}
+
+	if !path.IsAbs(etcdPath) {
+		panic(fmt.Sprintf("etcd-path %s is not absolute", etcdPath))
+	}
+
+	if err := restoreEtcd(); err != nil {
+		panic(err)
+	}
+}
+
+func restoreEtcd() error {
+	etcdCmd := exec.Command(etcdPath, "--force-new-cluster", "--data-dir", etcdRestoreDir)
+
+	etcdCmd.Stdout = os.Stdout
+	etcdCmd.Stderr = os.Stderr
+
+	if err := etcdCmd.Start(); err != nil {
+		return fmt.Errorf("Could not start etcd2: %s", err)
+	}
+	defer etcdCmd.Wait()
+	defer etcdCmd.Process.Kill()
+
+	return runCommands(10, 2*time.Second)
+}
+
+var clusterHealthRegex = regexp.MustCompile(".*cluster is healthy.*")
+var lineSplit = regexp.MustCompile("\n+")
+var colonSplit = regexp.MustCompile("\\:")
+
+func runCommands(maxRetry int, interval time.Duration) error {
+	var retryCnt int
+	for retryCnt = 1; retryCnt <= maxRetry; retryCnt++ {
+		out, err := exec.Command(etcdctlPath, "cluster-health").CombinedOutput()
+		if err == nil && clusterHealthRegex.Match(out) {
+			break
+		}
+		fmt.Printf("Error: %s: %s\n", err, string(out))
+		time.Sleep(interval)
+	}
+
+	if retryCnt > maxRetry {
+		return fmt.Errorf("Timed out waiting for healthy cluster\n")
+	}
+
+	memberID := ""
+	if out, err := exec.Command(etcdctlPath, "member", "list").CombinedOutput(); err != nil {
+		return fmt.Errorf("Error calling member list: %s", err)
+	} else {
```

Comment:
Prefer to skip the `else`, per [effective go](https://golang.org/doc/effective_go.html#if)